js/resultsPanel.js

```diff
@@ -0,0 +1,83 @@
+class ResultsPanel {
+    constructor(domContainer) {
+      this.container = domContainer;
+      this.generate();
+    }
+  
+    generate() {
+      this.resultsList = makeElement('div', { className: 'results-list' });
+      this.container.append(this.resultsList);
+    }
+  
+    clearAll() {
+      const removeItems = [...this.resultsList.children];
+      removeItems.forEach(li => li.remove());
+    }
+  
+    async addAllRes(data) {
+      let promAllLines = data.map(this.createResultLine);
+      const createdLines = await Promise.all(promAllLines);
```

Comment:
good usage of promise.all()! you can even optimize your code a little more if you ask for info about 3 companies in each http request